### DISCUSSION TITLE
Suggestions to improve the library by removing obsolete methods and classes

### DISCUSSION THREAD

**Discussion by ngoquoctoandev** [canton7/RestEase#260] [General]
I think you should delete all obsolete methods and classes. Your code has too much redundancy making it difficult to maintain and contribute to the project. Anyway thank you for the library is awesome too. In addition, I think it is much better to use the System.Text.Json library to reduce dependencies and focus on .NET 6 and above.

**Comment by canton7** [comment]
Unfortunately backwards compatibility is important - if you don't maintain backwards compatibility you annoy people and they go elsewhere. See #182 for an example.

For the other points, see #150.
